### PR TITLE
fix: Upgrade to typescript 4

### LINKED ISSUE #1769: Upgrade to Typescript 4
- Need to communicate a timeline

### PR DESCRIPTION
**Pull request by jasononeil**
# Objective

Get on the latest version of Typescript.
Fixes #1769 

# Motivation and Context

All consuming repos are using it. This will allow us to adopt some newer typescript features like in #1530  

# Changes

- Upgrade to the latest Typescript version
- Fix related errors, including:
    - fix: A fix to the `name` prop in TextAreaField which previously was not working, and Typescript now alerts us to this
    - docs: A fix to the design token storyboard docs to pass types
    - chore: A change to SplitButton types to use `DOMRect` instead of `ClientRect`. ClientRect is deprecated and DOMRect seems to be what we should be using now.
    - fix: In Illustration, cast caught errors to a string before creating an new Error() object.

# Checklist

- [ ] If this contains visual changes, has it been reviewed by a designer?
- [ ] If this introduces either a new component or a breaking change to an existing component, has it been reviewed by an advocate? (Ask in #prod_design_systems)
- [ ] I have considered likely risks of these changes and got someone else to QA as appropriate
- [ ] I have or will communicate these changes to the front end practice

# Additional Considerations
- Does there need to be right-to-left (RTL) options for localization and internationalization?
- Has the test suite been updated?
- Have you done cross-browser testing for our [supported browsers](https://academy.cultureamp.com/hc/en-us/articles/204539569-Supported-browsers-for-Participants)?
- Have you updated any relevant documentation or left useful comments in the code?
- Have you reviewed Culture Amp's Web Accessibility guide [Current Compliance, Going Forward, Statement and Answers for Customers](https://cultureamp.atlassian.net/wiki/spaces/Prod/pages/428572998/Web+Accessibility)?
- If this contains substantial visual changes, especially far reaching ones, have you unlocked the Chromatic step in the pipeline?
- Have Storybook stories been updated?

**If you're new to Kaizen, please ask #prod_design_systems to set up an onboarding session to get you up to speed.** If you have an urgent PR to merge before that happens, it is safest to ask in #prod_design_systems to have a design system advocate review the PR to catch any issues.

### CODE REVIEW

File: draft-packages/form/KaizenDraft/Form/TextAreaField/TextAreaField.tsx [hunk 1]
```diff
@@ -23,6 +23,7 @@ const TextAreaField: React.FunctionComponent<TextAreaFieldProps> = props => {
     validationMessage,
     status = "default",
     autogrow,
+    name,
```
Comment (ActuallyACat) [inline draft-packages/form/KaizenDraft/Form/TextAreaField/TextAreaField.tsx:26]: How was this not throwing an error before? :O

Comment (jasononeil) [reply draft-packages/form/KaizenDraft/Form/TextAreaField/TextAreaField.tsx:26]: This is my favourite: https://developer.mozilla.org/en-US/docs/Web/API/Window/name

Comment (jasononeil) [reply draft-packages/form/KaizenDraft/Form/TextAreaField/TextAreaField.tsx:26]: It's an implicit global, that in most browsers returns an empty string or null.

As of typescript v4, they've marked the variable as deprecated and type `void` to discourage its use and catch accidents like this. 

See lib.d.ts breaking changes here: https://devblogs.microsoft.com/typescript/announcing-typescript-4-0/#lib-d-ts-changes

**Review by ActuallyACat — APPROVED** [review]
Very exciting stuff! 🎉

File: draft-packages/form/KaizenDraft/Form/TextAreaField/TextAreaField.tsx (hunk 1 above)
Comment (ActuallyACat) [reply draft-packages/form/KaizenDraft/Form/TextAreaField/TextAreaField.tsx:26]: 😨😨😨 Good to know

**Pull request merged**